magprime/automated_emails.py

```diff
@@ -65,3 +65,15 @@ def __init__(self, event):
 
 StopsEmail('MAGFest Dept Checklist Introduction', 'dept_checklist_intro.txt',
            lambda a: a.is_single_dept_head and a.admin_account)
+
+"""
+TODO: change the condition of this to be when:
+- can_spam == True
+- today's date < 3 days before the deadline in the email of Dec 31st 2015
+- attendee registered more than 3 days ago (wait to send them this email til 3 days after they have registered)
+- attendee paid or part of a paid group (but not part of an unpaid group, like a waitlisted dealer)
+"""
+AutomatedEmail(Attendee, 'Last Chance for MAGFest 2016 bonus swag!', 'attendee_swag_promo.html',
+               lambda a: a.can_spam and
+                         a.paid == c.HAS_PAID and
+                         days_ago(3, a.registered))
```

Comment:
I'm fine either way, but in the interest of time, I just want to get this
thing out the door.

--
Nicholas Marinelli
Chief Operating Officer
MAGFest
http://magfest.org
On Dec 22, 2015 6:42 PM, "Eli Courtwright" <notifications@github.com> wrote:

> In magprime/automated_emails.py
> <https://github.com/magfest/magprime/pull/50#discussion_r48310979>:
>
> > @@ -65,3 +65,15 @@ def __init__(self, event):
> >
> >  StopsEmail('MAGFest Dept Checklist Introduction', 'dept_checklist_intro.txt',
> >             lambda a: a.is_single_dept_head and a.admin_account)
> > +
> > +"""
> > +TODO: change the condition of this to be when:
> > +- can_spam == True
> > +- today's date < 3 days before the deadline in the email of Dec 31st 2015
> > +- attendee registered more than 3 days ago (wait to send them this email til 3 days after they have registered)
> > +- attendee paid or part of a paid group (but not part of an unpaid group, like a waitlisted dealer)
> > +"""
> > +AutomatedEmail(Attendee, 'Last Chance for MAGFest 2016 bonus swag!', 'attendee_swag_promo.html',
> > +               lambda a: a.can_spam and
> > +                         a.paid == c.HAS_PAID and
> > +                         days_ago(3, a.registered))
>
> Your 1st and 3rd line look good. As for the paid, it should be something
> like
>
> (a.paid == c.HAS_PAID or a.group and a.group.amount_paid)
>
> We might also want a condition where we check whether they're under a
> certain donation level already. Like, if someone is already a Season
> Supporter, we probably don't need to send them this email since there's not
> much above that. At the very least, if someone has kicked in at the
> absolute highest level, we should probably not send this email.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/magfest/magprime/pull/50/files#r48310979>.
>